python/ray/serve/built_application.py

```diff
@@ -54,47 +54,42 @@ def __init__(self, deployments: List[Deployment]):
 
         self._deployments = ImmutableDeploymentDict(deployment_dict)
 
+        if ingress not in self._deployments:
+            raise ValueError(
+                f"Requested ingress deployment '{ingress}' was not found in the "
+                f"deployments passed in: {self._deployments.keys()}. Ingress must be "
+                "one of the deployments passed in."
+            )
+
+        self._ingress = ingress
+
     @property
     def deployments(self) -> ImmutableDeploymentDict:
         return self._deployments
 
     @property
     def ingress(self) -> Optional[Deployment]:
```

Comment:
what is the motivation we don't rely on this?

Might not be big deal. but `BuiltApplication` is public API right now. I think we can remove the decorator and hide from the public api.

Reply:
Do you mean why we don't rely on the previous implementation?

The `BuiltApplication` class only takes in a list of deployments, so there's nothing enforcing that only one deployment has a route prefix. Also, there's the possibility that no deployments have a route prefix if the user doesn't want their application exposed through HTTP. So I changed the class to explicitly take in an ingress deployment name, so everything is more well defined.